File: app/src/main/java/com/princeparadoxes/watertracker/ui/screen/main/statistic/StatisticAdapter.java
package com.princeparadoxes.watertracker.ui.screen.main.statistic;

import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.princeparadoxes.watertracker.R;

import java.util.ArrayList;
import java.util.List;

public class StatisticAdapter extends RecyclerView.Adapter<StatisticItemViewHolder> {

    private List<StatisticModel> data = new ArrayList<>();

    public StatisticAdapter() {
    }

    @Override
    public StatisticItemViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        LayoutInflater inflater = LayoutInflater.from(parent.getContext());
        View v = inflater.inflate(R.layout.statistic_type_view, parent, false);
        return new StatisticItemViewHolder(v);
    }

    @Override
    public void onBindViewHolder(StatisticItemViewHolder holder, int position) {
        holder.bindView(position, data.get(position), null);
    }

    @Override
    public int getItemCount() {
        return data.size();
    }

    public StatisticAdapter setData(List<StatisticModel> data) {
        this.data.clear();
        this.data.addAll(data);
        notifyDataSetChanged();
        return this;
    }
}

File: app/src/main/java/com/princeparadoxes/watertracker/presentation/screen/statistic/StatisticAdapter.kt
package com.princeparadoxes.watertracker.presentation.screen.statistic

import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup

import com.princeparadoxes.watertracker.R
import com.princeparadoxes.watertracker.domain.entity.StatisticModel
import com.princeparadoxes.watertracker.utils.toCalendar

import java.util.ArrayList

class StatisticAdapter : RecyclerView.Adapter<StatisticItemViewHolder>() {

    private val data = ArrayList<StatisticModel>()
    private lateinit var inflater: LayoutInflater

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): StatisticItemViewHolder {
        if (!this::inflater.isInitialized) inflater = LayoutInflater.from(parent.context)
        val view = inflater.inflate(R.layout.statistic_type_view, parent, false)
        return StatisticItemViewHolder(view)
    }

    override fun onBindViewHolder(holder: StatisticItemViewHolder, position: Int) {
        val w = 3L;
        w.toCalendar()
        holder.bindView(position, data[position], null)
    }

    override fun getItemCount(): Int = data.size

    fun getItem(position: Int) = data[position]

    fun setData(data: List<StatisticModel>): StatisticAdapter {
        this.data.clear()
        this.data.addAll(data)
        notifyDataSetChanged()
        return this
    }
}
